azure-iot-device/tests/iothub/shared_client_tests.py

```diff
@@ -530,6 +595,13 @@ def test_client_returned(
         assert client._mqtt_pipeline is mock_mqtt_pipeline_init.return_value
         assert client._http_pipeline is mock_http_pipeline_init.return_value
 
+    @pytest.mark.it("Raises a TypeError if the 'sastoken_ttl' kwarg is supplied by the user")
+    def test_sastoken_ttl(self, client_class, x509):
+        with pytest.raises(TypeError):
+            client_class.create_from_x509_certificate(
+                x509=x509, hostname=self.hostname, device_id=self.device_id, sastoken_ttl=1000
+            )
+
```

Comment:
There is a better pattern for handling excluded kwargs from a subset of factory methods, but I'll wait to implement it until we have more than one instance of this problem